src/Angor/Client/Services/MessageService.cs

```diff
@@ -0,0 +1,272 @@
+            IEncryptionService encryptionService,
+            INetworkService networkService,
+            IRelayService relayService,
+            NostrConversionHelper nostrHelper)
+        {
+            _logger = logger;
+            _nostrCommunicationFactory = nostrCommunicationFactory;
+            _encryptionService = encryptionService;
+            _networkService = networkService;
+            _relayService = relayService;
+            _nostrHelper = nostrHelper;
+        }
+
+        public void SetKeys(string currentUserPrivateKeyHex, string currentUserNpub, string contactHexPub)
+        {
+            _currentUserPrivateKeyHex = currentUserPrivateKeyHex;
+            _currentUserNpub = currentUserNpub;
+            _contactHexPub = contactHexPub;
+
+            if (!string.IsNullOrEmpty(_currentUserNpub))
+            {
+                _currentUserHexPub = _nostrHelper.ConvertBech32ToHex(_currentUserNpub);
+            }
+            else
+            {
+                _currentUserHexPub = null;
+            }
+        }
+
+        public async Task InitializeAsync(string currentUserPrivateKeyHex, string currentUserNpub, string contactHexPub)
+        {
+            _sinceTime = DateTime.UtcNow.AddDays(-7);
+
+            SetKeys(currentUserPrivateKeyHex, currentUserNpub, contactHexPub);
+
+            if (string.IsNullOrEmpty(_currentUserPrivateKeyHex) || string.IsNullOrEmpty(_contactHexPub) || string.IsNullOrEmpty(_currentUserHexPub))
+            {
+                _logger.LogWarning("InitializeAsync - Missing required keys.");
+                NotifyStateChanged();
+                return; 
+            }
+
+            IsLoadingMessages = true;
+            NotifyStateChanged();
+
+            try
+            {
+                await LoadMessagesAsync();
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError($"Failed to initialize messaging: {ex.Message}");
+            }
+            finally
+            {
+                IsLoadingMessages = false;
+                NotifyStateChanged();
+            }
+        }
+
+        public void DisconnectSubscriptions()
+        {
+            if (_subscriptionActive)
+            {
+                _relayService.DisconnectSubscription(_currentUserHexPub);
+                _relayService.DisconnectSubscription(_contactHexPub);
+
+                _sinceTime = DateTime.UtcNow.AddDays(-7);
+                _subscriptionActive = false;
+                _directMessagesDictionary.Clear();
+            }
+        }
+
+        public async Task LoadMessagesAsync()
+        {
+            if (string.IsNullOrEmpty(_currentUserPrivateKeyHex) || string.IsNullOrEmpty(_contactHexPub) || string.IsNullOrEmpty(_currentUserHexPub))
+            {
+                _logger.LogWarning("LoadNewMessagesAsync - Missing required keys.");
+                return;
+            }
+
+            try
+            {
+                DisconnectSubscriptions();
```

Comment:
Why disconnect?

Reply:
What is the question here? why to disconnect a subscription or why call it disconnect?